csharp/src/Ice/Communicator.cs

```diff
@@ -1297,6 +1306,39 @@ internal void UpdateConnectionObservers()
             }
         }
 
+        /// <summary>Activates the configured plug-ins.</summary>
+        /// <param name="cancel">The cancellation token.</param>
+        /// <returns>A task that completes once all the plug-ins are activated.</returns>
+        private async Task ActivatePluginsAsync(CancellationToken cancel = default)
+        {
+            // Invoke ActivateAsync on the plug-ins, in the order they were loaded.
+            var activatedPlugins = new List<IPlugin>();
+            try
+            {
+                foreach ((string name, IPlugin plugin) in _plugins)
+                {
+                    await plugin.ActivateAsync(cancel).ConfigureAwait(false);
+                    activatedPlugins.Add(plugin);
+                }
+            }
+            catch (Exception)
+            {
+                // Destroy the plug-ins that have been successfully activated, in the reverse order.
```

Comment:
I think we should add a TODO to just destroy the plug-ins during DisposeAsync instead.

Reply:
actually, we are calling twice DisposeAsync in each plug-in one here, and again in Communicator.DisposeAsync

Reply:
I just fixed it and removed this method.